example-widget-mui/src/InvitationsPage/InvitationsPage.tsx

```diff
@@ -0,0 +1,245 @@
+  Box,
+  Button,
+  Card,
+  CardContent,
+  CardHeader,
+  Chip,
+  CircularProgress,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  Stack,
+  TextField,
+} from '@mui/material';
+import { unstable_useId as useId } from '@mui/utils';
+import { isError } from 'lodash';
+import { MatrixCapabilities } from 'matrix-widget-api';
+import { ReactElement, useEffect, useState } from 'react';
+import { NavigationBar } from '../NavigationPage';
+
+/**
+ * A component that allows the user to query the user directory
+ * and invite others into the room.
+ */
+export const InvitationsPage = (): ReactElement => {
+  return (
+    <>
+      <NavigationBar title="Invitations" />
+
+      <Box my={1}>
+        <MuiCapabilitiesGuard
+          capabilities={[MatrixCapabilities.MSC3973UserDirectorySearch]}
+        >
+          <Box m={1}>
+            <InvitationsView />
+          </Box>
+        </MuiCapabilitiesGuard>
+      </Box>
+    </>
+  );
+};
+
+export const InvitationsView = (): ReactElement => {
+  const [term, setTerm] = useState('');
+  const [selected, setSelected] = useState<SearchResults>([]);
+  const [invitedUsers, setInvitedUsers] = useState<string[]>([]);
+
+  const { loading, results, error } = useUserSearchResults(term, 100);
+
+  const listTitleId = useId();
+
+  return (
+    <>
+      {error && (
+        <Alert severity="error" variant="outlined">
+          {error.message}
+        </Alert>
+      )}
+
+      <Stack direction="row" gap={1} my={1}>
+        <Autocomplete
+          sx={{ flex: 1 }}
+          getOptionLabel={(user) => user.userId}
+          // disable built-in filtering
+          filterOptions={(x) => x}
+          options={results}
+          includeInputInList
+          filterSelectedOptions
+          value={selected}
+          onChange={(_, value) => setSelected(value)}
+          onInputChange={(_, value) => {
+            setTerm(value);
+          }}
+          renderInput={(params) => (
+            <TextField
+              {...params}
+              label="Users"
+              size="medium"
+              InputProps={{
+                ...params.InputProps,
+                endAdornment: (
+                  <>
+                    {loading && <CircularProgress color="inherit" size={20} />}
+                    {params.InputProps.endAdornment}
+                  </>
+                ),
+              }}
+            />
+          )}
+          renderOption={(props, option) => (
+            <ListItem {...props}>
+              <ListItemIcon sx={{ mr: 1, minWidth: 0 }}>
+                <ElementAvatar
+                  userId={option.userId}
+                  displayName={option.displayName}
+                  avatarUrl={option.avatarUrl}
+                />
+              </ListItemIcon>
+
+              <ListItemText primary={option.displayName ?? option.userId} />
+            </ListItem>
+          )}
+          disablePortal
+          multiple
+          loading={loading || true}
```

Comment:
if we have loading || true. it will be always true or did i missed something?

Reply:
Oh yes. Good catch! That should've been removed after local testing.